### PR TITLE
ref(replay): Run geo info normalization on the replay user

### LINKED ISSUE #50639: Make geolocation tags searchable in Replays
### Problem Statement

The browser sdks populate the geolocation data for user and for node you can use setUser({}) to set user geolocation data like this - 

```
Sentry.setUser({
            id: "123455",
            ip_address: "33.12.2.4",
            email: "fran@gmail.com",
            geo: {
                country_code: "usa",
                city: "baltimore"
            }
})
```

When set, you can search for the properties in Discover.  Since replays use setUser({}) as well, we should make these properties also searchable for Replay.

### Solution Brainstorm

Attach the geolocation data to replay events, or at least allow the geo.* tags to be searchable in the Replay search bar.

### Product Area

Replays

### PR DESCRIPTION
**Pull request by Dav1dde**
Fixes: https://github.com/getsentry/sentry/issues/50639

Additionally refactors the normalization code a bit to bundle a lot of the arguments that are passed around in a single struct. Also updates the feature flag checks to use `should_filter` instead of `has_feature` to make it work on Proxy Relays.

### CODE REVIEW

File: relay-server/src/services/processor/replay.rs [hunk 1]
```diff
@@ -118,29 +109,32 @@ pub fn process(
     Ok(())
 }
 
+#[derive(Debug)]
+struct ReplayState<'a> {
```
Comment (jjbayer) [inline relay-server/src/services/processor/replay.rs]: nit: We have similar structs for event and span normalization, where we call it `*Config`. Should we rename this to `ReplayProcessingConfig` or similar?

File: relay-server/src/services/processor/replay.rs [hunk 2]
```diff
@@ -118,29 +109,32 @@ pub fn process(
...
+    pub config: &'a ProjectConfig,
+    pub global_config: &'a GlobalConfig,
+    pub geoip_lookup: Option<&'a GeoIpLookup>,
+    pub event_id: Option<EventId>,
+    pub project_id: Option<ProjectId>,
+    pub organization_id: Option<u64>,
+    pub client_addr: Option<IpAddr>,
+    pub user_agent: RawUserAgentInfo<String>,
```
Comment (jjbayer) [inline relay-server/src/services/processor/replay.rs:123]: What does `pub` do on a field of a private struct?

File: relay-server/src/services/processor/replay.rs [hunk 3]
```diff
@@ -24,91 +24,82 @@ use crate::statsd::{RelayCounters, RelayTimers};
...
-    if replays_disabled {
+    if state.should_filter(Feature::SessionReplay) {
         state.managed_envelope.drop_items_silently();
         return Ok(());
     }
 
     // If the replay video feature is not enabled check the envelope items for a
     // replay video event.
-    if replay_video_disabled && count_replay_video_events(state) > 0 {
+    if state.should_filter(Feature::SessionReplayVideoDisabled)
```
Comment (jjbayer) [inline relay-server/src/services/processor/replay.rs]: Good catch!

**Review by jjbayer — APPROVED** [review]

File: relay-server/src/services/processor/replay.rs [hunk 4]
```diff
@@ -277,48 +269,23 @@ struct ReplayVideoEvent {
...
+    state: &ReplayState<'_>,
 ) -> Result<Bytes, ProcessingError> {
     let ReplayVideoEvent {
         replay_event,
         replay_recording,
         replay_video,
-    } = match rmp_serde::from_slice(&payload) {
-        Ok(result) => result,
-        Err(e) => {
-            relay_log::warn!("replay-video-event: {e} {event_id:?}");
```
Comment (loewenheim) [inline relay-server/src/services/processor/replay.rs:300]: Is this log intentionally gone?

File: relay-server/src/services/processor/replay.rs (hunk 2 above)
Comment (Dav1dde) [reply relay-server/src/services/processor/replay.rs:123]: Just 'documentation' that these fields can be freely accessed as opposed to stuff which should only be exposed via methods.

File: relay-server/src/services/processor/replay.rs (hunk 4 above)
Comment (Dav1dde) [reply relay-server/src/services/processor/replay.rs:300]: Yes. I removed it, it seemed like a leftover debugging thing, it contains almost no information and it is logged at a level which it shouldn't be. This is just broken user input which should not result in a warning, at most it's a debug info.

**Pull request merged**